### ISSUE TITLE
ProductsCommandHandler implementation

### ISSUE THREAD

**Issue by pwasilewski-netcode** [MicrosoftDocs/architecture-center#1801]
Hello,
ProductsCommandHandler implements more than one ICommandHandler&lt;T&gt;, that breaks "Single responsibility rule" from SOLID.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0c0ec92e-c237-f3a6-1835-f2078486dcb3
* Version Independent ID: f597c9a2-f18f-3722-57f9-acd535472b0b
* Content: [Command and Query Responsibility Segregation (CQRS) pattern - Cloud Design Patterns](https://docs.microsoft.com/en-us/azure/architecture/patterns/cqrs#feedback)
* Content Source: [docs/patterns/cqrs.md](https://github.com/microsoftdocs/architecture-center/blob/master/docs/patterns/cqrs.md)
* Service: **architecture-center**
* Sub-service: **cloud-fundamentals**
* GitHub Login: @dragon119
* Microsoft Alias: **pnp**

**Comment by KetanChawda-MSFT** [comment]
@pwasilewski-netcode Thanks for the feedback! I have assigned the issue to the content author @MikeWasson  to evaluate and update as appropriate.

**Comment by MikeWasson** [comment]
I think the "single responsibility" in this case is "handling commands related to product inventory" - you could further decompose it, but then you'd need an additional mechanism to dispatch the messages to the right handler. In fact ProductsCommandHandler could just be a dispatcher that invokes other code modules.

**Comment by pwasilewski-netcode** [comment]
I realised that there is no command dispatcher so it's ok to simplify the case in that way.
In the other hand i think that a point of view depends on scale of responsibility, class that add product to the inventory shouldn't be able to rate the product. Besides I guess that inventory has own repository.
According to Uncle Bob "There should be never more than one reason for a class to change." and the other quote "remember that the reasons for change are people".
So, to sum up it depends. We're both right :) Thanks for a chat.
